ReactiveCocoa/ControlSubject.swift

```diff
@@ -0,0 +1,44 @@
+import ReactiveSwift
+import enum Result.NoError
+
+/// Represents a subject of an interactive control.
+///
+/// Generally speaking, `ControlSubject` emits changes only for the user
+/// interactions in which the subject specializes.
+public final class ControlSubject<Value> {
```

Comment:
@sharplet The trouble boils down to multiple "sources of truth" in one's code. Ultimately we want a UI control to reflect a value that we've set somewhere in the (View)Model, but we rarely/never want to initialize our model using the current value of a control.

Yes, the *implementation* of a control certainly requires the storage of its own value so that it can be redrawn, replaced dynamically, etc. But we _should not_ expect that the control contains _any_ truth at all and hence should not be asked to store values on our behalf. It is a simple contract to fulfill: just display the value I ask you to, and notify me when the user makes changes to that value.

In contrast, to expose a "proper property" then we are asking the control to let us know if anything programmatically changes that value as well as the current/initial value of the control. In order to do that, it must also behave like a "proper property" in that our setting of the property's value directly should propagate back to us.

The alternative to the above was to make a "kinda-sorta property" that didn't fulfill the expectations of all the other Properties in the framework, which was a huge red flag in my eyes.

(Also, what @andersio said above.)